### DISCUSSION TITLE
Get an instance of the Vue application in a custom module

### DISCUSSION THREAD

**Discussion by dimmkan** [directus/directus#14221] [Archive]
Good afternoon. Could you tell me how to get an instance of the Vue application in a custom module that I'm making? I need to install the plugin before I use other components inside the module. How is it possible to do this in a similar way as it would be in a normal application:

```
importing {createApp} from 'vue';

const app = create app(app);
app.use(MyPlugin);
```

**Comment by rijkvanzanten** [comment]
There's currently no access to the app context. What plugin are you trying to instantiate? Most modern Vue plugins allow was to just import components/directives/composables without having to inject on the global scope

**Reply by dimmkan** [reply]
I want to attach 'primevue/config', since some imported components do not get the desired configuration from the application instance, for example Calendar or Dropdown.

**Reply by dimmkan** [reply]
I would like to make an analogue of this design:

```
import {createApp} from 'vue';
import App from './App.vue';
import PrimeVue from 'primevue/config';
const app = createApp(App);

app.use(PrimeVue);
```

**Reply by rijkvanzanten** [reply]
Extensions currently don't have access to the `app` instance, as the app is already mounted by the time extensions are downloaded and registered.

**Reply by dimmkan** [reply]
> Extensions currently don't have access to the `app` instance, as the app is already mounted by the time extensions are downloaded and registered.

It's sad, but thank you very much for the answer. Is it possible to count on the fact that in the future such an opportunity will appear?

**Reply by rijkvanzanten** [reply]
Sure, but I can't make any promises if/when that might happen
